packages/redux-store/src/utils.ts

```diff
@@ -22,7 +22,7 @@ interface AgentThunkApiConfig {
 
 function createAsyncAgentThunk<Returned, ThunkArg = void>(
   typePrefix: string,
-  payloadCreator: AsyncThunkPayloadCreator<Returned, ThunkArg, AgentThunkApiConfig>
+  payloadCreator: AsyncThunkPayloadCreator<Promise<Returned>, ThunkArg, AgentThunkApiConfig>
```

Comment:
How does this suddenly need a promise?

Reply:
This type is not required. https://github.com/reduxjs/redux-toolkit/blob/c111bba464dd53b4af743b86da53e5de9098ca5a/packages/toolkit/etc/redux-toolkit.api.md?plain=1#L168 wraps the return type in a `MaybePromise` which uses `PromiseLike` which takes care of this. I also tested it locally and it does work all good with `yarn check-types`.

Reply:
Yes, it does work. Seems to be a quirk from from my editor's EsLint